pkg/scheduler/cache/cache_test.go

```diff
@@ -344,3 +355,121 @@ func TestBindTasks(t *testing.T) {
 		t.Fatalf("succesfully binding task should have 1 event")
 	}
 }
+
+var resourceKey = schema.GroupVersionResource{
+	Group:    "example.io",
+	Version:  "v1",
+	Resource: "foo",
+}
+
+type testResource struct {
+	name     string
+	sc       *SchedulerCache
+	ch       chan struct{}
+	handlers CustomResourceEventHandlerFuncs
+}
+
+func (t *testResource) Name() string {
+	return t.name
+}
+
+func (t *testResource) EventHandlerFuncs(gvr schema.GroupVersionResource) kcache.ResourceEventHandler {
+	return t.handlers[gvr]
+}
+
+func newHandler(ch chan struct{}, sc *SchedulerCache) (CustomResourceEventHandler, error) {
+	t := &testResource{
+		name:     "test-handler",
+		sc:       sc,
+		ch:       ch,
+		handlers: make(CustomResourceEventHandlerFuncs),
+	}
+
+	t.handlers[resourceKey] = kcache.ResourceEventHandlerFuncs{
+		AddFunc: t.onResourceAdd,
+	}
+	return t, nil
+}
+
+func (t *testResource) onResourceAdd(obj interface{}) {
+	if _, ok := t.sc.CustomResources[resourceKey]; !ok {
+		t.sc.CustomResources[resourceKey] = make(map[string]schedulingapi.CustomResource)
+	}
+
+	t.sc.Mutex.Lock()
+	defer t.sc.Mutex.Unlock()
+
+	t.sc.CustomResources[resourceKey]["name-foo"] = nil
+	t.ch <- struct{}{}
+	klog.InfoS("Added resource")
+}
+
+func newUnstructured(apiVersion, kind, namespace, name string) *unstructured.Unstructured {
+	return &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": apiVersion,
+			"kind":       kind,
+			"metadata": map[string]interface{}{
+				"namespace": namespace,
+				"name":      name,
+			},
+		},
+	}
+}
+
+func Test_add_custom_event_handler(t *testing.T) {
+	gvrToListKind := map[schema.GroupVersionResource]string{
+		{Group: "example.io", Version: "v1", Resource: "foo"}: "fooList",
+	}
+	ch := make(chan struct{})
+	tests := []struct {
+		name        string
+		sc          *SchedulerCache
+		handlerName string
+		creator     CustomResourceEventHandlerCreator
+		trigger     func(gvr schema.GroupVersionResource, client dynamic.Interface) error
+		want        map[schema.GroupVersionResource]map[string]schedulingapi.CustomResource
+	}{
+		{
+			name:        "custom resource add",
+			handlerName: "test",
+			sc: &SchedulerCache{
```

Comment:
How about use `NewDefaultMockSchedulerCache` to trigger the mock list/watch process?

Reply:
It's not a proper way here, because custom clinet and cache is a independent part, and it needs to provide parameters separately, and NewDefaultMockSchedulerCache  is a general function for test.